R/controls.R

```diff
@@ -0,0 +1,38 @@
+#' @param html the content of the control. May be provided as string or as
+#'   HTML generated with Shiny/htmltools tags
+#' @param position position of control: 'topleft', 'topright',
+#'   'bottomleft', or 'bottomright'
+#' @param controlId the id of this control
+#' @param classes extra CSS classes to append to the control
+#'
+#' @describeIn map-layers Add arbitrary HTML controls to the map
+#' @export
+addControl = function(map,
+    html,
+    position=c('topleft', 'topright', 'bottomleft', 'bottomright'),
+    controlId=NULL,
+    classes=c('info', 'legend'),
+    data=getMapData(map)) {
```

Comment:
Just thought more about it. Let's keep this argument. There might be cases in which we construct `html` using `data`, although I don't have an example on my mind right now.